### DISCUSSION TITLE
http 405 response after a POST on an API route

### DISCUSSION THREAD

**Discussion by gabooh** [vercel/community#1365] [Help] [answered]
When submitting an html form from a SSG page, that submits to an API route (/api/order), I get an http 405 error.
The code of the API route is very simple :

export default (req, res) => {
  console.log(req.body)

  res.redirect('/confirmOrder')
}

The content of the post http request is correctly logged, I can see it in the Vercel dashboard, so the request is made and processed. The redirect seams to work as the browser reports the new URL, with with a 405 response, not the rendered page.
/confirmOrder is a static page.
It works locally in dev mode and after a next build and next start. But not on Vercel.

I have no vercel.json, but it doesn't look like I need one.
I tried a few things but I have no idea any more what could I need to change.
Any idea ? Thank you

**Comment by gabooh** [comment, marked as answer]
Ok I've found the solution after letting it sink for a moment ... I need a 303 redirect as I'm redirecting to a route that needs a GET http request. And the default 307 redirect doesn't change the method. And Vercel is strict so this didn't work.

`res.redirect(303, '/confirmOrder')`

Could have been easier to debug with an error message in the dashboard : "route XXX doesn't support POST request" or something like that.

**Reply by fedpre** [reply]
Thank you I had that problem as well and it was driving me crazy!

**Reply by neo7-studio-web** [reply]
Wow thank you ! 
After a long search I was finally able to find your (simple) solution for a Vercel serverless function.

This sould be in the doc @vercel-team

**Reply by amyegan** [reply]
Getting a ticket created for it now. Thanks for highlighting this!

**Reply by gojira-tech** [reply]
Thank you so much, it helped me to resolve the issue !
